etl/src/main/java/org/pih/hivmigration/etl/sql/ExamLabResultsMigrator.groovy

```diff
@@ -72,6 +76,72 @@ class ExamLabResultsMigrator extends ObsMigrator {
+            SELECT
+                he.source_encounter_id,
+                concept_uuid_from_mapping('PIH', '1435'),
+                CASE
+                    WHEN result LIKE 'negative%' THEN concept_uuid_from_mapping('PIH', 'NEGATIVE')
+                    WHEN result LIKE 'positive%' THEN concept_uuid_from_mapping('PIH', 'POSITIVE')
+                    END
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE
+                lab_test = 'ppd'
+                AND result LIKE 'negative%' OR result LIKE 'positive%';
+        ''')
+
+        executeMysql("Migrate numeric PPD", '''
+            INSERT INTO tmp_obs (source_encounter_id, concept_uuid, value_numeric)
+            SELECT
+                he.source_encounter_id,
+                concept_uuid_from_mapping('PIH', 'TUBERCULIN SKIN TEST'),
+                TRIM(result)
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE lab_test = 'ppd' AND is_number(result);
+        ''')
+
+        executeMysql("Log warning about invalid PPD values", '''
+            INSERT INTO hivmigration_data_warnings (patient_id, encounter_id, field_name, field_value, note)
+            SELECT
+                hp.person_id,
+                he.encounter_id,
+                'HIV_EXAM_LAB_RESULTS ppd',
+                helr.result,
+                'Invalid value. Not migrated.'
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN hivmigration_patients hp on he.source_patient_id = hp.source_patient_id
+            WHERE lab_test = 'ppd' AND result NOT LIKE 'negative%' AND result NOT LIKE 'positive%' AND NOT is_number(result);
+        ''')
+
+        executeMysql("Log warning about missing encounters", '''
+            INSERT INTO hivmigration_data_warnings (patient_id, encounter_id, field_name, field_value, note)
+            SELECT
+                hp.person_id,
+                he.encounter_id,
+                'HIV_EXAM_LAB_RESULTS ppd',
+                helr.result,
+                CONCAT('Belongs to an encounter that was not migrated, source ID ', he.source_encounter_id, '. Not migrated')
+            FROM hivmigration_exam_lab_results helr
+            JOIN hivmigration_encounters he on helr.source_encounter_id = he.source_encounter_id
+            JOIN hivmigration_patients hp on he.source_patient_id = hp.source_patient_id
+            LEFT JOIN encounter e on he.encounter_id = e.encounter_id
+            WHERE lab_test = 'ppd' AND e.encounter_id IS NULL;
+        ''')
```

Comment:
Long-term, it may be okay to just have a data warning that compares the hivmigration_encounters table to the encounter table and log any encounters that weren't migrated?